x-pack/plugins/security_solution/common/endpoint/types/actions.ts

```diff
@@ -98,10 +118,14 @@ export type EndpointActionDataParameterTypes =
   | undefined
   | ResponseActionParametersWithPidOrEntityId;
 
-export interface EndpointActionData<T extends EndpointActionDataParameterTypes = undefined> {
+export interface EndpointActionData<
+  T extends EndpointActionDataParameterTypes = undefined,
+  TOutputActions extends OutputActions = OutputActions
```

Comment:
`OutputAction` is really not conforming to `{ type: 'json' | 'text' ; content: Record }` because you could define a type with only `type` and I think this would accepted it. Here is what I would suggest:


```typescript
export interface EndpointActionData<
  TParams extends EndpointActionDataParameterTypes = undefined,
  TOutputContent extends object = object> {

  output?: {
    type: 'json' | 'text',
    content: TOutputContent
  }
}
```

With this, you should be able to:

```
declare action: EndpointActionData<never, { entries: RunningProcessesEntry[] }>;
```

I think you are trying to also limit the `type` for each type of action, but that's probably not necessary because its unlikely that an action response might support multiple types of output. 


Also - I noticed that `T` is set to `undefined`. Probably should hav been set to `never` by default. I don't remember what PR introduced that.